tuf/signed/verify.go

```diff
@@ -57,18 +56,18 @@ func VerifyRoot(s *data.Signed, minVersion int, keys map[string]data.PublicKey)
 			continue
 		}
 		// threshold of 1 so return on first success
-		return verifyMeta(s, "root", minVersion)
+		return verifyMeta(s, data.CanonicalRootRole, minVersion)
 	}
 	return ErrRoleThreshold{}
 }
 
 // Verify checks the signatures and metadata (expiry, version) for the signed role
 // data
-func Verify(s *data.Signed, role string, minVersion int, db *keys.KeyDB) error {
-	if err := verifyMeta(s, role, minVersion); err != nil {
+func Verify(s *data.Signed, role *data.RoleWithKeys, minVersion int) error {
+	if err := verifyMeta(s, role.Name, minVersion); err != nil {
 		return err
 	}
-	return VerifySignatures(s, role, db)
+	return VerifySignatures(s, role)
```

Comment:
It was previously called by some unused code, which I deleted.  I thought `Verify` would probably go away at some point because verifying the metadata would be moved to `tuf.Repo.CheckAndSet_*`, and those would then call `VerifySignatures` and we can remove `Verify` entirely.

I'm happy to begin the merging process now, but it seems like it would be temporarily strangely named (since `VerifySignatures` would be verifying the metadata as well as the signatures)